### PR TITLE
[Feat|DETS] : Preconfigured connector URL fixes and unit tests

### PR DESCRIPTION
**Pull request by amoldashwant**
### Fixed
- Preconfigured connector URL '/' issue
- Dynamic asset name generation issue

### Added
- Unit tests for Controller and service

## Pre-review checks

Please ensure to do as many of the following checks as possible, before asking for committer review:

- [x] DEPENDENCIES are up-to-date. [Dash license tool](https://github.com/eclipse/dash-licenses). Committers can open IP issues for restricted libs.
- [x] [Copyright and license header](https://eclipse-tractusx.github.io/docs/release/trg-7/trg-7-02) are present on all affected files

### CODE REVIEW

File: src/main/resources/edc-request-template/sample-asset.json [hunk 1]
```diff
@@ -5,10 +5,10 @@
 		"@id": "sample",
 		"properties": {
 			"id": "sample",
-			"name": "Welcome to Catena-X",
+			"name": "%s",
 			"contenttype": "application/json",
 			"policy-id": "use-eu",
-			"type": "data.core.digitalTwin.submodel"
+			"type": "data.core.sample"
```
Comment (sachinargade123) [inline src/main/resources/edc-request-template/sample-asset.json]: Use 'type' dynamically from properties file like name attribute.

Comment (amoldashwant) [reply src/main/resources/edc-request-template/sample-asset.json]: Code changes are done. PR updated

**Review by sachinargade123 — APPROVED** [review]
LGTM

**Review by almadigabor — CHANGES_REQUESTED** [review]
You need to adjust the license headers in the files you modified in this PR to include 2024 like [here](https://eclipse-tractusx.github.io/docs/release/trg-7/trg-7-02#copyright-and-license-header) as well.

**Comment by amoldashwant** [comment]
@almadigabor '2024' change added in the header

**Review by almadigabor — APPROVED** [review]
LGTM

**Pull request merged**